### PR TITLE
chore(junit/assert): adjust junit and assert documentation

### PR DESCRIPTION
**Pull request by mboskamp**
Related to [CAM-14460](https://jira.camunda.com/browse/CAM-14460)

### CODE REVIEW

File: content/update/minor/716-to-717/_index.md [hunk 1]
```diff
@@ -203,3 +205,57 @@ To migrate from 7.16 to 7.17, and from H2 1.4 to 2.0, follow these steps:
...
+## JUnit 5 Maven coordinates
+
+When migrating to the new version of the JUnit 5 extension, make sure to update the maven coordinates of the dependency.
+The group id changed from `<groupId>org.camunda.bpm.extension</groupId>` to `<groupId>org.camunda.bpm</groupId>` and the versioning schema is now tied to Camunda Platform 7. The first release of the new JUnit 5 extension will have the version `7.17.0`.
+
+```xml
+    <dependency>
+      <groupId>org.camunda.bpm</groupId>
+      <artifactId>camunda-bpm-junit5</artifactId>
+      <version>7.17.0</version>
```
Comment (tasso94) [inline content/update/minor/716-to-717/_index.md:227]:
```suggestion
      <version>7.17.0</version>
      <scope>test</scope>
```

Aren't the example coordinates more complete with the scope assigned to `test`?

**Review by tasso94 — CHANGES_REQUESTED** [review]
Looks good to me in general. However, I have some suggestions and improvement ideas.

File: content/user-guide/testing/_index.md [hunk 2]
```diff
@@ -17,11 +17,105 @@ This section explains how to write unit tests and integration tests with Camunda
...
+The Camunda Platform provides helper classes to write unit tests for JUnit versions 3, 4 and 5.
+
+## JUnit 5
+
+Camunda Platform version 7.17.0+ ships with a {{< javadocref page="org/camunda/bpm/engine/test/junit5/ProcessEngineExtension.html" text="JUnit 5 extension" >}} that provides access to the process engine and services through getter methods.
+
+The extensions process engine is configured by the default configuration file called `camunda.cfg.xml` which needs to be placed on the classpath. A custom configuration file can be passed to the extension when creating the ProcessEngineExtension object.
+
+We did not include the JUnit 5 extension with the `camunda-engine` artifact like we did with the JUnit 4 {{< javadocref page="org/camunda/bpm/engine/test/ProcessEngineRule.html" text="ProcessEngineRule" >}}. The main reason is that Camunda Platform is
+using JUnit 4 for testing the engine and other components and adding JUnit 5 into the mix would complicate the test setup.
```
[the remark this answers is not in the document]
Comment (mboskamp) [reply content/user-guide/testing/_index.md:29]:
```suggestion
```

**Comment by mboskamp** [comment]
Thanks, @tasso94 for the thorough review. All your suggestions made sense to me so I included them. Are we ready to merge then? :)

**Review by tasso94 — APPROVED** [review]
🥳

**Pull request merged**